dev/devicelab/lib/framework/apk_utils.dart

```diff
@@ -83,6 +83,83 @@ bool hasMultipleOccurrences(String text, Pattern pattern) {
   return text.indexOf(pattern) != text.lastIndexOf(pattern);
 }
 
+
+/// Utility class to analyze the content inside an APK.
+class ApkExtractor {
+  ApkExtractor(this.apkFile);
+
+  /// The APK.
+  final File apkFile;
+
+  bool _extracted = false;
+
+  Directory _outputDir;
+
+  Future<void> _extractApk() async {
+    if (_extracted) {
+      return;
+    }
+    _outputDir = await apkFile.parent.createTemp('apk');
+    await eval('unzip', <String>[apkFile.path, '-d', _outputDir.path]);
+    _extracted = true;
+  }
+
+  Future<String> _findDexDump() async {
+    final String androidHome = Platform.environment['ANDROID_HOME'] ??
+        Platform.environment['ANDROID_SDK_ROOT'];
+
+    if (androidHome == null || androidHome.isEmpty) {
+      throw 'Unset env flag: `ANDROID_HOME` or `ANDROID_SDK_ROOT`.';
+    }
+
+    final String dexdumps = await eval('find', <String>[androidHome, '-name', 'dexdump']);
```

Comment:
nit: you can pass `-quit` to find and it will just give you the first result.

Reply:
Same caveat about this not working on Windows.

Could we instead expect dexdumps to be at a specific path in the SDK for a given version of build tools?

Reply:
`-quit` didn't seem to work. :/ .    

> Could we instead expect dexdumps to be at a specific path in the SDK for a given version of build tools?

That is possible, however it'd be nice if we can detect the version automatically.  `AndroidSdk` does this. It looks through the directories and then extracts the available version SDK from the directory name.

Reply:
This should work on Windows now.